### PR TITLE
fix firing TU cost display in TB mode

### PR DESCRIPTION
**Pull request by FranciscoDA**
Display the TU cost to fire when hovering over a hostile unit in the battlescape in turn-based mode, like in vanilla apocalypse

### CODE REVIEW

**Comment by FilmBoy84** [comment]
Looks fantastic to me; play-tests well
Code checks with my limited knowledge all pass
Are we good to merge @JonnyH ?

File: game/ui/tileview/battletileview.cpp [hunk 1]
```diff
@@ -847,57 +848,76 @@ void BattleTileView::render()
...
+								else if (drawPathPreview)
 								{
 									sp<Image> img;
-									switch (calculatedAttackCost)
+									switch (previewedPathCost)
 									{
-										case -4:
-											img = nullptr;
-											break;
 										case -3:
```
[the remark this answers is not in the document]
Comment (FranciscoDA) [reply game/ui/tileview/battletileview.cpp]: Would it be OK to have an enum of negative values for those special cases and simply casting positives integers to the enum type? Like so
```
enum class PreviewedPathType : int {
   UNREACHABLE = -3,
   TOO_FAR = -2,
   NOT_STORED = -1
}
PreviewedPathType previewedPathCost;
```
Then an actually valid path could be static_cast'ed `previewedPathCost = static_cast<PreviewedPathType>(5);`. This isn't undefined behaviour as long as the storage type is specified: https://en.cppreference.com/w/cpp/language/enum

[the remark this answers is not in the document]
Comment (JonnyH) [reply game/ui/tileview/battletileview.cpp]: Yup, that sounds good to me.

Or another option would be just use an int, and have
'constexpr int PATH_UNREACHABLE = -1;'
etc. in the tightest possible scope (this class?)

I don't really mind which, whatever you feel is best, but I just don't want "-1" to mean something without a /name/

[the remark this answers is not in the document]
Comment (FranciscoDA) [reply game/ui/tileview/battletileview.cpp]: Fair enough, this is solved in dd557dc, albeit with a ton of const_casts :P

File: game/ui/tileview/battletileview.h [hunk 2]
```diff
@@ -107,22 +108,26 @@ class BattleTileView : public TileView
 	sp<Image> pathPreviewUnreachable;
 	std::list<Vec3<int>> pathPreview;
 	int pathPreviewTicksAccumulated = 0;
-	// -3 = unreachable
-	// -2 = too far
-	// -1 = no previewed path stored
-	// 0+ = path cost
-	int previewedPathCost = -1;
+	enum class PreviewedPathCostSpecial : int
```
Comment (JonnyH) [inline game/ui/tileview/battletileview.h]: You're right, the lots of static_casts<> here are pretty ugly.

I'm then leaning towards something like:
constexpr int PREVIEW_PATH_UNREACHABLE = -1;
etc.

Thoughts?

Comment (JonnyH) [reply game/ui/tileview/battletileview.h]: ..but this is getting really nit-picky. The core idea (No magic numbers) has been solved. We can always change it later anyway

**Pull request merged**